### PR TITLE
the http2RoundTripper should only use http2

### LINKED ISSUE #896: TLS follow-up: determine why http2Round only transport still supports http1
The http2RoundTripper will speak protocols other than http 2 (e.g. http1). 

It shouldn't. 

Please fix this. If it turns out that this outcome cannot be realized because we are misunderstanding what support client-go provides on this subject, then let's discuss how we should change the code to better work with that library.

Follow-up from #820

https://github.com/vmware-tanzu/pinniped/blob/cd686ffdf31ae2a5f7241cd8d3ddf304c0684a19/internal/concierge/impersonator/impersonator.go#L474

### PR DESCRIPTION
**Pull request by margocrawf**
```release-note
Fixed bug where http2-only client was also allowing http1.1.
```

Fixes #896

### CODE REVIEW

File: internal/concierge/impersonator/impersonator.go [hunk 1]
```diff
@@ -812,6 +813,15 @@ func getTransportForProtocol(restConfig *rest.Config, protocol string) (http.Rou
 		return nil, fmt.Errorf("could not build transport: %w", err)
 	}
 
+	// For clients that support http2, transport.New calls http2.ConfigureTransports,
+	// which configures with both h2 and http/1.1,
+	// even when you explicitly only ask for h2.
+	// Override that change.
+	cfg, err := apimachinerynet.TLSClientConfig(rt)
```
Comment (enj) [inline internal/concierge/impersonator/impersonator.go:820]: This import is already in this file:

```suggestion
	cfg, err := utilnet.TLSClientConfig(rt)
```

File: internal/concierge/impersonator/impersonator.go [hunk 2]
```diff
@@ -812,6 +813,15 @@ func getTransportForProtocol(restConfig *rest.Config, protocol string) (http.Rou
...
+	if err != nil {
+		return nil, err
```
Comment (enj) [inline internal/concierge/impersonator/impersonator.go:822]:
```suggestion
		return nil, fmt.Errorf("could not extract TLS config: %w", err)
```

**Review by enj — CHANGES_REQUESTED** [review]
Nice work :)

File: internal/concierge/impersonator/impersonator.go (hunk 1 above)
Comment (margocrawf) [reply internal/concierge/impersonator/impersonator.go:820]: Whoops! I totally missed that somehow!

**Review by enj — APPROVED** [review]

**Pull request merged**